### ISSUE TITLE
Extension stopped working

### ISSUE THREAD

**Issue by acristoffers** [tridactyl/tridactyl#5103]
-   Brief description of the problem:

Today I turned on the PC and Tridactyl is not working. It's like it is always in insert mode, even
though it is in normal mode, so none of the shortcuts work. Sometimes `:` work, but not always.

-   Tridactyl version (`:version`):

1.24.1

-   Firefox version (Top right menu > Help > About Firefox):

133.0

-   URL of the website the bug happens on:

All sites.

-   Config:

https://gist.github.com/acristoffers/dd159074f4a0d0f29f31088a985b970a

-   Contents of ~/.tridactylrc or ~/.config/tridactyl/tridactylrc:

```
" General Settings
set configversion 2.0
set theme midnight
set smoothscroll true
set searchengine brave
set hintfiltermode vimperator
set hintchars asdfjkl;gh
set update.lastchecktime 1663051397130
set searchurls.g https://www.google.com/search?q=
set searchurls.gs https://scholar.google.com/scholar?q=
set searchurls.b https://www.bing.com/search?q=
set searchurls.ddg https://duckduckgo.com/?q=
set searchurls.y https://search.yahoo.com/search?p=
set searchurls.t https://twitter.com/search?q=
set searchurls.w https://en.wikipedia.org/wiki/Special:Search/
set searchurls.y https://www.youtube.com/results?search_query=
set searchurls.a https://www.amazon.fr/s?k=
set searchurls.startpage https://startpage.com/do/search?language=english&cat=web&prfe=375fb2a5e70e8e61721e2f3b10b93a14f24d547dca5803bda4d8e03b70e0588e0e4d21e36deeccba529df4ac942f071d6ec0905f7723f70e933aafae8120c9b908fd73fa16545e71d6a065ce&query=
set searchurls.ghs https://github.com/search?utf8=✓&q=
set searchurls.yt https://www.youtube.com/results?search_query=
set searchurls.gh https://github.com/
set searchurls.nix https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=
set searchurls.brave https://search.brave.com/search?q=
set searchurls.vg redacted

set searchurls.an redacted
set searchurls.ani redacted
set searchurls.ap redacted
set searchurls.ans redacted
set searchurls.aps redacted

set searchurls.da redacted
set searchurls.cl redacted
set searchurls.clt redactedjk
set searchurls.r  redacted

set searchurls.pan redacted
set searchurls.pap redacted
set searchurls.pans redacted
set searchurls.paps redacted

" For syntax highlighting see https://github.com/tridactyl/vim-tridactyl
" vim: set filetype=tridactyl
```

**Comment by acristoffers** [comment]
I had `g` and `o` trigger correctly, but it happened once and then they wouldn't work again. I guess
something is messing with the keybindings. I started a private window, where tridactyl is enabled
but most of my extensions are not and the problem exists there too.

**Comment by acristoffers** [comment]
The only other extension running in private mode is Bitwarden, so I doubt it is interfering somehow.

**Comment by bovine3dom** [comment]
Could you check if it works in a fresh profile? `about:profiles` -> create a new profile -> launch profile in new browser -> install tridactyl

**Comment by acristoffers** [comment]
I deleted .Mozilla and started fresh, no difference.

**Comment by bovine3dom** [comment]
I'm not 100% sure that's the same as making a new profile, could you check?

And also check Tridactyl stable Vs beta?

**Comment by acristoffers** [comment]
It is the same thing since the profile folder lives there, but before trying that I also tried just
creating a new profile and installing the extension. I also tried with the beta version. Nothing
works.

I also just saw that shift+esc always works, and toggles between normal and ignore. The j key also
works as expected, but not k. I tried enabling debug mode and watching the console, and the
extension is not getting any input when I press o, f, g, etc, but does get and process w, j, esc and :.

**Comment by acristoffers** [comment]
I don't know if it helps, but it's Fedora 41, up-to-date. The Firefox is the one from dnf, but I
also tried the one from the flatpak and it has the same issue.

**Comment by acristoffers** [comment]
The problem is only happening in this Fedora box. At home I tried in NixOS and even though I have
the same versions and configuration, I don't see the problem there. I'll try to figure out what is
wrong. When I have the time I'll also install Fedora in a virtual machine in my home pc to see it happens there too.

**Comment by acristoffers** [comment]
The following error is on one Firefox console (`ctrl+shift+j`):

```
Unexpected pre-existing callContextData on ""Tridactyl" (ID: tridactyl.vim@cmcaine.co.uk, moz-extension://9a3ccc84-88ae-4048-935c-b4e58b89d02c/)" contextId 30 ExtensionParent.sys.mjs:657
    withCallContextData resource://gre/modules/ExtensionParent.sys.mjs:657
    result resource://gre/modules/ExtensionParent.sys.mjs:1222
    withPendingBrowser resource://gre/modules/ExtensionParent.sys.mjs:676
    result resource://gre/modules/ExtensionParent.sys.mjs:1221
    callAndLog resource://gre/modules/ExtensionParent.sys.mjs:1172
    recvAPICall resource://gre/modules/ExtensionParent.sys.mjs:1220
    AsyncFunctionNext self-hosted:800
    permitUnload chrome://global/content/elements/browser-custom-element.js:1867
    _beginRemoveTab chrome://browser/content/tabbrowser/tabbrowser.js:4444
    removeTab chrome://browser/content/tabbrowser/tabbrowser.js:4328
    remove chrome://browser/content/parent/ext-tabs.js:857
    result resource://gre/modules/ExtensionParent.sys.mjs:1223
    withCallContextData resource://gre/modules/ExtensionParent.sys.mjs:666
    result resource://gre/modules/ExtensionParent.sys.mjs:1222
    withPendingBrowser resource://gre/modules/ExtensionParent.sys.mjs:676
    result resource://gre/modules/ExtensionParent.sys.mjs:1221
    callAndLog resource://gre/modules/ExtensionParent.sys.mjs:1172
    recvAPICall resource://gre/modules/ExtensionParent.sys.mjs:1220
    AsyncFunctionNext self-hosted:800
```

**Comment by acristoffers** [comment]
I finally figured it out and it is the weirdest behavior I have ever seen. It was Gnome's screen
reader accessibility thing that somehow turned on. I have no idea how that works, because I did not
see any on-screen difference, but it was totally soaking my key-presses in Firefox, except that not
always. I tried an online page that just dumps the contents of the key-press events and saw that even
with no extensions I was not getting most of the events. This made me try another browser, and it
also had the issue. It was totally by accident that I saw the accessibility icon in the bar, and
after disabling it the site started registering all key-presses again and the extension started
working again. So for anyone having this kind of issue in the future: look for accessibility stuff
that got enabled without you noticing. I did not see the functionality mess with any other
application I use, only the browsers, but this is possibly because they implement support and the
other apps don't.
